app/src/main/java/com/kodeco/android/coordplot/components/PlotSurface.kt

```diff
@@ -0,0 +1,106 @@
+package com.kodeco.android.coordplot.components
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Slider
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.kodeco.android.coordplot.ui.theme.MyApplicationTheme
+import java.math.RoundingMode
+import java.text.DecimalFormat
+
+@Composable
+fun PlotSurface() {
+    var xPercentage: Float by rememberSaveable { mutableStateOf(0.5f) }
+    var yPercentage: Float by rememberSaveable { mutableStateOf(0.5f) }
+
+    fun getRounded(value: Float) : String? {
+        val format = DecimalFormat("#.#")
+        format.roundingMode = RoundingMode.DOWN
+        return format.format(value)
+    }
+
+    Surface(
+        modifier = Modifier.fillMaxSize(),
+        color = Color.White
+    ) {
+        Column(
```

Comment:
Here, you can create a separate MapSlider composable flle and pass the values (such as title, percentage, lambda expression) as parameter to the composable method. In this way we can reduce the repetition of the code. 
Also try to handle the code on Landscape orientation as well.

Reply:
Done!, will check the changes for handle landscape mode